app/controllers/posts_controller.rb

```diff
@@ -1,6 +1,7 @@
 class PostsController < ApplicationController
   def index
-    @user_id = User.find params.require(:user_id)
+    @user_id = User.find(params[:user_id])
+    # @user_id = User.find params.require(:user_id)
```

Comment:
- [ ] Please delete the unused comment.
- [ ] For solve the n+1 problem you should write `posts = @user.posts.includes(:comments)`   `@posts = @user.posts`

in your case you are using `@user_posts = Post.where(author_id: @user_id)` so you can add `includes(comments)`
[Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec_n%2B1.md#:~:text=Make%20sure%20that%20the%20N%2B1%20problem%20is%20solved%20when%20fetching%20all%20posts%20and%20their%20comments%20for%20a%20user.%20This%20is%20the%20corresponding%20wireframe)